core/web/transfer_controller.go

```diff
@@ -40,9 +41,52 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
-	db := tc.App.GetSqlxDB()
-	q := pg.NewQ(db, tc.App.GetLogger(), tc.App.GetConfig())
-	etx, err := bulletprooftxmanager.SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("withdrawal source address is missing: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		balance := chain.BalanceMonitor().GetEthBalance(tr.FromAddress)
+
+		if balance == nil || balance.IsZero() {
+			jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("balance is too low for this transaction to be executed: %v", balance))
+			return
+		}
+
+		var gasPrice *big.Int
+
+		gasLimit := chain.Config().EvmGasLimitTransfer()
+		estimator := chain.TxManager().GetGasEstimator()
+
+		if chain.Config().EvmEIP1559DynamicFees() {
+			gasPrice = chain.Config().EvmMaxGasPriceWei()
+			_, gasLimit, err = estimator.GetDynamicFee(gasLimit)
+			if err != nil {
+				jsonAPIError(c, http.StatusUnprocessableEntity, errors.Wrap(err, "failed to get dynamic gas fee"))
+				return
+			}
+		} else {
+			gasPrice, gasLimit, err = estimator.GetLegacyGas(nil, gasLimit)
+			if err != nil {
+				jsonAPIError(c, http.StatusUnprocessableEntity, errors.Wrap(err, "failed to estimate gas"))
+				return
+			}
+		}
+
+		intBalance := balance.ToInt()
+		fee := gasPrice.Mul(gasPrice, utils.NewBigI(int64(gasLimit)).ToInt())
+
+		intBalance = intBalance.Add(intBalance, fee)
+
+		// ETH balance is less than the sent amount
+		if intBalance.Cmp(tr.Amount.ToInt()) < 0 {
```

Comment:
Is this checking `balance + fee > amount` instead of `balance > amount + fee`?
Or perhaps `balance - fee > amount` was intended:
```suggestion
		intBalance = intBalance.Sub(intBalance, fee)

		// ETH balance is less than the sent amount
		if intBalance.Cmp(tr.Amount.ToInt()) < 0 {
```

Reply:
We are here validating the balance + the fees is enough for the amount being tried to be transferred.